### PR TITLE
Core/BanManager: Rewrite Ban system

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
* Added ban manager
* Delete ban functions in world class
* Seperate ban account to (ban account, ban ip, ban character)
* Support world announce

##### TESTS PERFORMED:
- [x] Test in game
- [x] Test in build MSVS 2019
 
##### HOW TO TEST THE CHANGES:
1. Apply commit and sql update file
2. Try writing the command to ban 
2.1. `.ban account`
2.2. `.ban character`
2.3. `.ban playeraccount`
2.4. `.ban ip`
3. If you set conf option 
3.1. `ShowBanInWorld = 1` everyone will see a notification about a ban
3.2. `ShowBanInWorld = 0` everything should be as before

##### KNOWN ISSUES AND TODO LIST:

- [x] Test in build Travis
- [ ] Test in game

##### Target branch(es): Master

### CODE REVIEW

File: src/server/game/Misc/BanManager.cpp [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    if (sWorld->getBoolConfig(CONFIG_SHOW_BAN_IN_WORLD))
+    {
+        bool IsPermanetly = true;
+
+        if (TimeStringToSecs(Duration) > 0)
+            IsPermanetly = false;
+
+        if (IsPermanetly)
+            sWorld->SendWorldText(LANG_BAN_IP_YOUPERMBANNEDMESSAGE_WORLD, Author.c_str(), IP.c_str(), Reason.c_str());
```
Comment (Viste) [inline src/server/game/Misc/BanManager.cpp]: Cant see this in Miscellaneous/Language.h

Comment (Winfidonarleyan) [reply src/server/game/Misc/BanManager.cpp]: Okay, i fixed (maybe) :D

File: src/server/game/Misc/BanManager.cpp [hunk 2]
```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
```
Comment (FrancescoBorzi) [inline src/server/game/Misc/BanManager.cpp]: the new code should be released under AGPL

File: src/server/game/Misc/BanManager.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
```
Comment (FrancescoBorzi) [inline src/server/game/Misc/BanManager.h]: same here :)

File: src/server/worldserver/worldserver.conf.dist [hunk 4]
```diff
@@ -3015,6 +3015,15 @@ Guild.AllowMultipleGuildMaster = 0
...
 ShowKickInWorld = 0
 
+#
+#     ShowBanInWorld
+#        Description: Determines whether a message is broadcasted to the entire server when a
+#                     player gets banned.
+#        Default:     0 - (Disabled)
+#                     1 - (Enabled)
+
+ShowBanInWorld = 1
```
Comment (FrancescoBorzi) [inline src/server/worldserver/worldserver.conf.dist]: it says default 0 but then it's 1 :D

**Comment by FrancescoBorzi** [comment]
This PR also makes the code more modular. Good job @Winfidonarleyan !

File: src/server/game/Misc/BanManager.cpp (hunk 2 above)
Comment (Winfidonarleyan) [reply src/server/game/Misc/BanManager.cpp]: Okay

File: src/server/worldserver/worldserver.conf.dist (hunk 4 above)
Comment (Winfidonarleyan) [reply src/server/worldserver/worldserver.conf.dist]: Ah, yes

**Comment by poszer** [comment]
Nice @Winfidonarleyan

**Comment by Winfidonarleyan** [comment]
@FrancescoBorzi, I summon you :D. Please check license

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan the license is good now, now need to wait for testers feedback :)

**Comment by Winfidonarleyan** [comment]
Perfectly

**Comment by Viste** [comment]
Tested all ban commands works as should

**Comment by Viste** [comment]
also can be tested here http://server-test.azerothcore.org/ for gm access write me in Discord

**Comment by BarbzYHOOL** [comment]
you should add the random duration  ban function winfi :P

**Pull request merged**